clang/lib/Sema/SemaStmtAttr.cpp

```diff
@@ -207,6 +174,46 @@ Sema::BuildSYCLIntelFPGAIVDepAttr(const AttributeCommonInfo &CI, Expr *Expr1,
       SYCLIntelFPGAIVDepAttr(Context, CI, SafeLenExpr, ArrayExpr, SafelenValue);
 }
 
+template <typename FPGALoopAttrT>
+FPGALoopAttrT *Sema::BuildSYCLIntelFPGALoopAttr(const AttributeCommonInfo &A,
+                                                Expr *E) {
+
+  unsigned SafeInterval = 0;
+  Expr *SafeExpr = nullptr;
+  if (E->isInstantiationDependent()) {
+    SafeExpr = E;
+  } else {
+    llvm::APSInt ArgVal(32);
+
+    if (!E->isIntegerConstantExpr(ArgVal, getASTContext())) {
+      Diag(E->getExprLoc(), diag::err_attribute_argument_type)
+          << (A.getParsedKind() == ParsedAttr::AT_SYCLIntelFPGAII
```

Comment:
Instead, see if you can just use the 'getSpelling' from the attribute to get its actual spelling.